### PR TITLE
Core: Pipeline - Separate unit tests from build job

### PR DESCRIPTION
**Pull request by junglebus**

### CODE REVIEW

**Comment by junglebus** [comment]
Hold off from merging this
I'm seeing the tests only take one minute to run (i.e. not slowing the total pipeline time), but the time after posting the results to completing the job is ridiculous. Might be transient Azure issue.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -250,6 +254,58 @@ stages:
...
+    jobs:
+    - job: UnitTests
+      workspace:
+        clean: all
+      strategy:
+        matrix:
+          Mono:
+            buildDescription: Mono
+            imageName: ubuntu-latest
+            framework: net461
```
Comment (ngosang) [inline azure-pipelines.yml]: Could you add net461 as variable?

Comment (junglebus) [reply azure-pipelines.yml]: I disagree. .NET full framework is end of life and this won't change. If we make this a variable, ubuntu-latest should a variable etc.

**Comment by junglebus** [comment]
Ready

**Pull request merged**